drivers/timer/stm32_lptim_timer.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2018 Foundries.io Ltd
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <drivers/timer/system_timer.h>
+#include <sys_clock.h>
+
+#include <spinlock.h>
+
+/*
+ * Assumptions and limitations:
+ *
+ * - system clock based on an LPTIM1 instance, clocked by LSI or LSE
+ * - prescaler is set to 1 (LL_LPTIM_PRESCALER_DIV1 in the related register)
+ * - using LPTIM1 AutoReload capability to trig the IRQ (timeout irq)
+ * - when timeout irq occurs the counter is already reset
+ * - the maximum timeout duration is reached with the LPTIM_TIMEBASE value
+ * - with prescaler of 1, the max timeout (LPTIM_TIMEBASE) is 2seconds
+ */
+
+#define LPTIM_CLOCK CONFIG_STM32_LPTIM_CLOCK
+#define LPTIM_TIMEBASE CONFIG_STM32_LPTIM_TIMEBASE
+
+/* nb of LPTIM counter unit per ms */
+#define COUNT_PER_MS (LPTIM_CLOCK / 1000)
+
+/* nb of LPTIM counter unit per kernel tick  */
+#define COUNT_PER_TICK (LPTIM_CLOCK / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static u32_t last_cnt;
+static u32_t accumulated_lptim_ms;
```

Comment:
Seems to be equal to 49 days -- quite okay but wrap-around will happen sooner that later in a real product ... moving to u64_t would make it less likely to happen

Reply:
we changed to store the total nb of LPTIM count unit, which is shorter.